### ISSUE TITLE
ManifoldSAE REML: continuation pre-warm always fails (β length 0 != decoder dim), cold-solves every seed

### ISSUE THREAD

**Issue by SauersML** [SauersML/gam#577]
## Summary

In the `ManifoldSAE` REML solve (`gamfit._sae_manifold` / `SaeManifoldOuterObjective`),
the **continuation pre-warm fails on every seed** and falls back to a full cold
solve each time. This makes `ManifoldSAE.fit()` pay full solve cost `N_seeds`
times and is the dominant runtime cost — minutes for a toy problem.

## Log evidence

```
[OUTER] SAE manifold: continuation pre-warm for seed N did not complete
  (SaeManifoldOuterObjective::seed_inner_state: β length 0 != decoder dim 6144
   (continuation stuck at oversmooth offset 2.772589e1, final rho dim=14));
  falling back to a cold seed eval
```

This message repeats for seed 0, 1, 2, … — i.e. the warm-start is **never**
used. The tell is `β length 0 != decoder dim`: the inner state's `β` is empty
when the continuation tries to seed it, so the pre-warm bails and we cold-solve.

## Impact

- `ManifoldSAE.fit()` over ~1.5k–6k rows at toy scale (D=32–64, ~6–10 atoms,
  intrinsic_rank=2, n_basis=10–12) runs for **minutes** on CPU; a 6000-row fit
  did not finish within 2 min. Pure-backprop training (`fit_every=0`) is fast;
  the cost is entirely in the closed-form REML solve's repeated cold seeds.

## Likely cause / ask

`seed_inner_state` expects a populated `β` to warm-start the continuation but
receives length 0, so every seed degrades to cold. Either:
1. Seed `β` correctly from the prior continuation step (so pre-warm completes), or
2. Skip the doomed pre-warm attempt when `β` is empty rather than logging +
   falling back (avoids the wasted continuation work), and/or
3. Expose `n_seeds` / a fast-path tolerance via `RemlConfig` so small problems
   don't sweep many seeds.

## Repro

`SauersML/Manifold-SAE` `experiments/orthogonal_manifolds.py` at tiny scale
(D=32, 4 manifolds + 2 slack atoms, intrinsic_rank=2, n_basis=10), calling
`sae.fit(x)`. See SauersML/Manifold-SAE#1.

## Environment
gamfit 0.1.141, torch 2.12.0, CPU (CUDA not compiled in), macOS arm64.

**Comment by SauersML** [comment]
Constructive follow-up: the **lower-level per-curve REML path fits the same kind
of data in ~20 ms with no continuation/seed sweep**, which suggests the SAE
solver could fall back to (or be built on) it.

```python
import numpy as np, torch, time, gamfit
torch.set_default_dtype(torch.float64)
rng = np.random.default_rng(0); N = 800
t = rng.random(N); u = 2*t - 1
intr = np.stack([u, u**2 - 1/3], -1)              # parabola
Q, _ = np.linalg.qr(rng.standard_normal((16, 2)))
y = torch.tensor(intr @ Q.T + 0.01*rng.standard_normal((N, 16)))

s = time.time()
out = gamfit.gaussian_reml_fit_positions(torch.tensor(t), y, basis="duchon",
                                         basis_order=2)
print("time %.3fs  R2 %.4f  lambda %.2e"
      % (time.time()-s,
         (1 - ((y-out["fitted"])**2).sum()/((y-y.mean(0))**2).sum()).item(),
         out["lambda"]))
# -> time 0.019s  R2 0.9962  lambda 3.5e-06
```

Same ambient dim, same Duchon m=2 basis, REML-selected smoothing — but it returns
in 19 ms with an analytic-backward cache, versus `ManifoldSAE.fit()` /
`sae_manifold_fit` spending minutes on the continuation pre-warm + cold-seed
sweep (this issue) and then hard-failing on ill-conditioning (#578).

Per-atom this is exactly the curve fit the SAE needs; the batched variant
`gaussian_reml_fit_positions_batched` packs all atoms ragged in one call (modulo
the `row_offsets` dtype papercut, #581). Would it be feasible for the SAE outer
solve to delegate per-atom curve fits to this path, or at least fall back to it
when the continuation seed sweep fails validation, instead of raising?

**Comment by SauersML** [comment]
## Fixed (the fatal rejection / cold-fallback)

`SaeManifoldOuterObjective::seed_inner_state` violated the documented continuation
contract (`src/solver/reml/continuation.rs:727`: "an empty-β seed means 'no
warm-start available, use your own cold default' and MUST be accepted as a
no-op"). It errored on `beta.len() == 0`, which is exactly what the continuation
pre-warm forwards before the first accepted eval (`state.last_beta` starts
empty) — dropping every continuation seed.

Fix: empty β → `Ok(SeedOutcome::NoSlot)` (proceed cold, no error). The SAE eval
already publishes `inner_beta_hint: Some(beta_hat)`, so step 1 cold-solves and
steps 2+ warm-start normally.

Verified: the `β length 0 != decoder dim` rejection spam is gone and the seed
sweep no longer aborts.

Caveat: this removes the fatal rejection, but `sae.fit()` is still slow at toy
scale — the dominant cost is the Efs outer solve itself, not the warm-start
wiring. The constructive fast-path suggestion above (delegate per-atom curve fits
to the 19 ms `gaussian_reml_fit_positions`) still stands as the real
runtime fix.

**Comment by SauersML** [comment]
🔒 claim: claude-code coordinator session — taking this now. (cluster: sae-reml-core)

**Comment by SauersML** [comment]
## Still impractically slow in gamfit 0.1.144 — scaling benchmark

`sae_manifold_fit` (the same Efs joint solver behind `ManifoldSAE.fit`) remains
unusably slow for real manifold recovery. Wall-clock on tiny problems, gamfit
0.1.144, CPU, macOS arm64:

```
n=150  K=2  d_atom=1  n_iter=8 :   4.0 s
n=300  K=3  d_atom=1  n_iter=8 :  21.6 s   (~5x for 2x points + 1 atom)
n=400  K=4  d_atom=1  n_iter=30:  > 180 s  (killed)
```

Super-linear in n, and steep in K and n_iter. A practical manifold-recovery run
(thousands of tokens, several atoms) is many minutes-to-hours, so the joint solver
can't be used as the recovery engine despite being the only *general* (any-shape,
position-optimizing) path.

For contrast, a decoupled recovery on the same data — K-subspaces clustering for
assignment + per-atom `gaussian_reml_fit_positions` for the curve — recovers all
manifolds in **~0.5 s for n=1500, K=4** (3000x+ faster), because each piece is a
closed-form / few-iteration solve. The joint Efs outer loop is the bottleneck.

Repro:
```python
import gamfit, numpy as np, time
from manifold_sae.data_synthetic import SyntheticDataset  # any orthogonal-curves generator
for n,K in [(150,2),(300,3)]:
    X = SyntheticDataset(d_ambient=12,n_features=K,n_samples=n,sparsity=0.4,noise=0.02,
                         seed=0,orthogonal_subspaces=True,curve_indices=[1,9,8][:K]).x.numpy().astype(float)
    t=time.time(); gamfit.sae_manifold_fit(X,K=K,d_atom=1,n_iter=8,random_state=0)
    print(n,K,round(time.time()-t,1),"s")
```

The fast-path delegation suggested earlier in this thread (per-atom curve fits via
`gaussian_reml_fit_positions`, positions held fixed within an outer assignment
loop) would close most of this gap.
